nominal/core/datasource.py

```diff
@@ -142,6 +149,50 @@ def get_write_stream(
             clients=self._clients,
         )
 
+    @overload  # type: ignore[misc]
+    def get_read_stream(
+        self,
+        format: Literal["pandas"],
+        *,
+        points_per_request: int = DEFAULT_POINTS_PER_REQUEST,
+        points_per_batch: int = DEFAULT_POINTS_PER_BATCH,
+        channels_per_request: int = DEFAULT_CHANNELS_PER_REQUEST,
+        num_workers: int = 1,
+        pool_type: PoolType = DEFAULT_POOL_TYPE,
+    ) -> ExportStream[pd.DataFrame]: ...
+
+    def get_read_stream(
```

Comment:
I'm hesitant to name this as a "stream" as it's not really using a streaming API.